### DISCUSSION TITLE
Python results do not match up to native command

### DISCUSSION THREAD

**Discussion by fidesachates** [coqui-ai/STT#2103] [Q&A] [answered]
Hi,

I'm just starting my journey into using coqui so I might be missing obvious things. The gist is I downloaded the latest release of coqui's native client and tested it against a few recorded wav files. They results were perfect. I then put them against a python server I "wrote" (mostly copied from [here](https://github.com/coqui-ai/STT-examples/blob/r1.0/python_websocket_server/stt_server/engine.py)), the results are terrible. 

What have I missed? Is the expectation that the python version match the native client a bad assumption on my part?

**Comment by fidesachates** [comment, marked as answer]
I think I answered my own question... the server code works when I re-encode my wave files to be mono channel 16k. I think this means that the native client is doing the re-encoding behind the scenes.

**Reply by reuben** [reply]
Yes, it is! We also have some code to do the same in the command-line driver that gets installed with the Python package, you can see it here: https://github.com/coqui-ai/STT/blob/main/native_client/python/client.py

The code for the native `stt` command line driver is here: https://github.com/coqui-ai/STT/blob/main/native_client/client.cc
